sagemaker-fundamentals/execution-role/execution-role.ipynb

```diff
@@ -0,0 +1,588 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# View AmazonSageMakerFullAccess policy\n",
+    "sagemaker_full = iam.get_policy(\n",
+    "    PolicyArn='arn:aws:iam::aws:policy/AmazonSageMakerFullAccess')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "versions = iam.list_policy_versions(\n",
+    "    PolicyArn='arn:aws:iam::aws:policy/AmazonSageMakerFullAccess')\n",
+    "pp.pprint(versions['Versions'][0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The latest version for `AmazonSageMakerFullAccess` policy is `v19`. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# view the latest version of AmazonSageMakerFullAccess policy\n",
+    "pp.pprint(\n",
+    "    iam.get_policy_version(\n",
+    "        PolicyArn='arn:aws:iam::aws:policy/AmazonSageMakerFullAccess',\n",
+    "        VersionId='v19'\n",
+    "    )\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "For most use cases of Amazon SageMaker, the `AmazonSageMakerFullAccess` policy would be sufficient. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Attach AmazonSageMakerFullAccess to basic-role\n",
+    "\n",
+    "res = iam.attach_role_policy(\n",
+    "    RoleName=role_res['Role']['RoleName'],\n",
+    "    PolicyArn='arn:aws:iam::aws:policy/AmazonSageMakerFullAccess',\n",
+    ")\n",
+    "\n",
+    "pp.pprint(res)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Clean up\n",
+    "\n",
+    "If you only plan to use the role to run certain application once, then it is a good practice to delete the role. If you plan to delete a role, make sure **you have detached all policies associated with it** and you **do not** have any Amazon EC2 instances running with the role you are about to delete. Deleting a role or instance profile that is associated with a running instance will break any applications running on the instance.\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# detach attached policies\n",
+    "attached_policies = iam.list_attached_role_policies(RoleName='basic-role')['AttachedPolicies']\n",
+    "for p in attached_policies:\n",
+    "    iam.detach_role_policy(\n",
+    "        RoleName=role_res['Role']['RoleName'],\n",
+    "        PolicyArn=p['PolicyArn']\n",
+    "    )\n",
+    "\n",
+    "# delete the role\n",
+    "res = iam.delete_role(\n",
+    "   RoleName=role_res['Role']['RoleName']\n",
+    ")\n",
+    "\n",
+    "pp.pprint(res)"
+   ]
+  }
+ ],
+ "metadata": {
```

Comment:
In fact, any kernel can work as long as it has boto3 installed. Since this nb is intended to be run on ec2 or user's local machine, I did not recommend a sagemaker-notebook-instance-styled kernel